test-network-function/platform/suite.go

```diff
@@ -441,6 +441,8 @@ func testTainted(env *config.TestEnvironment) {
 					// Looks through the accepted taints listed in the tnf-config file.
 					// If all of the tainted modules show up in the configuration file, don't fail the test.
 					nodeTaintsAccepted = taintsAccepted(env.Config.AcceptedKernelTaints, taintedModules)
+				} else if len(individualTaints) > 0 {
```

Comment:
What if we have "module was loaded" plus other not-related to out-of-tree module taints? The white list mechanism would allow those taints to pass this TC.

Reply:
True.  I'll play around with it more.

Reply:
Okay, I added counters to always mark the node as `failed` if there are other tainted reasons compared to `module was loaded`.